src/evict.c

```diff
@@ -722,7 +727,20 @@ int performEvictions(void) {
                 }
             }
         } else {
-            goto cant_free; /* nothing to free... */
+            if (server.maxmemory_soft_scale) {
+                break;
+            } else {
+                goto cant_free; /* nothing to free... */
+            }
+        }
+    }
+    if (server.maxmemory_soft_scale) {
+        size_t mem_used = zmalloc_used_memory();
+        size_t overhead = freeMemoryGetNotCountedMemory();
+        mem_used = (mem_used > overhead) ? mem_used - overhead : 0;
+        if (mem_used < server.maxmemory_soft) {
+            result = EVICT_OK;
+            goto update_metrics;
```

Comment:
I think we can simplify the logic and check the more fundamental conditions as follows (and remove L747 too). Note that `mem_used` comes from L549. Thoughts?
```suggestion
            break;
        }
    }
    
    if (mem_used - mem_freed < server.maxmemory) {
        result = EVICT_OK;
    } else if (isEvictionProcRunning) {       
        result = EVICT_RUNNING;
    } else {
        result = EVICT_FAIL;
    }
```